CHANGELOG.md

```diff
@@ -2,9 +2,62 @@
 
 ## [Unreleased]
 
-### Fixed
+### Changed
 
+- Also apply `empty_item_single_line=true` to trait definitions [#5047](https://github.com/rust-lang/rustfmt/issues/5047)
+- Replace `structopt` dependency with `clap` [PR #5239](https://github.com/rust-lang/rustfmt/pull/5239)
+
+### Fixed
+
+- Block indent line breaks for type alias impl traits (TAITs) when `version = "Two"` [#5027](https://github.com/rust-lang/rustfmt/issues/5027)
+- Retain trailing comments in module when using `#![rustfmt::skip]` [#5033](https://github.com/rust-lang/rustfmt/issues/5033)
+- Remove trailing whitespace when formatting a where clause with an empty right hand side [#5012](https://github.com/rust-lang/rustfmt/issues/5012) [#4850](https://github.com/rust-lang/rustfmt/issues/4850)
+- Fix various module resolution issues:
+  - Handle external mods imported via external->inline load hierarchy [#5063](https://github.com/rust-lang/rustfmt/issues/5063)
+  - Resolve sub modules of integration tests [#5119](https://github.com/rust-lang/rustfmt/issues/5119)
+  - Module resolution will fallback to the current search directory if a relative directory search results in a `FileNotFound` error [#5198](https://github.com/rust-lang/rustfmt/issues/5198)
+  - Give clearer error messsaeg when a module is found in two places (e.g `x.rs` and `x/mod.rs`) [#5167](https://github.com/rust-lang/rustfmt/issues/5167)
+- Prevent rustfmt from adding an `impl Trait` definition into types [#5086](https://github.com/rust-lang/rustfmt/issues/5086)
+- Correctly format associated type in macro body [#4823](https://github.com/rust-lang/rustfmt/issues/4823)
+- Fix cases where `normalize_comments=true` would de-normalizes some comments [#4909](https://github.com/rust-lang/rustfmt/issues/4909)
+- Prevent rustfmt from wrapping reference style links [#5095](https://github.com/rust-lang/rustfmt/issues/5095) and [#4933](https://github.com/rust-lang/rustfmt/issues/4933)
+- Prevent rustfmt from always adding an empty comment line when formatting itemized blocks [#5088](https://github.com/rust-lang/rustfmt/issues/5088)
+- Don't format files annotated with an inner `#![rustfmt::skip]` attribute [PR #5094](https://github.com/rust-lang/rustfmt/pull/5094)
+- Remove duplicate comma when struct pattern ends with `..` and `trailing_comma=Always` [#5066](https://github.com/rust-lang/rustfmt/issues/5066)
+- Fix static async closure qualifier order [#5149](https://github.com/rust-lang/rustfmt/issues/5149)
```

Comment:
This seems like a good place to make note of a general observation I'm having. Note that I'm not asking for any changes here, just sharing a thought for us to bounce around going forward around how we structure things.

If I'm a user perusing this list, many of these don't really tell me much, and I need to click in (at least once, if not more) to get to a location that has more details which I then need to digest, and sometimes the linked issue or PR doesn't have much information either.

I think it would be nice in the future if we can articulate these in such a way that a Rust dev could read and say "ah okay I understand what the problem was and/or what the change fixed" without having to do that subsequent deep dive. However, that could be better suited for the release notes type of model if we decide to split as I noted above.

Reply:
I'm totally open to going back and making edits! I'm sure I can better articulate some of these as well. I agree that a lot of them don't give you a full understanding of the issue and the fix. I think I was trying to limit myself to a one sentence description, but some of these might need a little more than that.

On the topic of internal changes, I remember you mentioned the idea of introducing an Internal changes section. I think it would be nice to include and I wonder if we could follow a similar model as rust-analyzer. See their [changelog-127](https://rust-analyzer.github.io/thisweek/2022/05/02/changelog-127.html) for reference.

Sticking with the rust-analyzer changelog just a little longer, I also like what they include images. rustfmt's output is also visual so we might be able to do something like that too. Overkill for some issues, but maybe for some we could include a side by side comparison of how rustfmt formatted a code snippet before and after the fix (maybe using two side by side GIFs??).

To help people more easily grok the changes maybe we could also try to logically group issues into categories based on the fix or the issue label we've assigned it? for example, grouping all the `a-comment` related fixes together in their own subsection under the `### Fixed` subheading.

Reply:
That certainly sounds intriguing and helpful! The released-but-not-documented changes/fixes/enhancements/etc. are really starting to stack up on us though, so let's try to get things caught up within our current model and continue the dialog on how to make things better going forward